HW9/task_1.py

```diff
@@ -0,0 +1,123 @@
+            self.__workers_quantity = new_workers_quantity
+        else:
+            print(f'Workers quantity should be a int or float with length equal or less than 6:'
+                  f' {type(new_workers_quantity)} : {len(new_workers_quantity)}')
+
+    @workers_quantity.deleter
+    def workers_quantity(self):
+        raise NotImplementedError('Delete is not supported for this attribute')
+
+    @property
+    def productivity_percent_of_department(self):
+        return self.__productivity_percent_of_department
+
+    @productivity_percent_of_department.setter
+    def productivity_percent_of_department(self, new_results_of_department):
+        if isinstance(new_results_of_department, int | float) and len(str(new_results_of_department)) <= 7:
+            self.__productivity_percent_of_department = new_results_of_department
+        else:
+            print(f'Results of department should be a int or float with length equal or less than 7:'
+                  f' {type(new_results_of_department)} : {len(new_results_of_department)}')
+
+    @productivity_percent_of_department.deleter
+    def productivity_percent_of_department(self):
+        raise NotImplementedError('Delete is not supported for this attribute')
+
+    @property
+    def total_dep_exp(self):
+        return self.__total_dep_exp
+
+    @total_dep_exp.setter
+    def total_dep_exp(self, new_total_exp):
+        if isinstance(new_total_exp, int | float) and len(str(new_total_exp)) <= 9:
+            self.__total_dep_exp = new_total_exp
+        else:
+            print(f'Total experience of department should be a int or float with length equal or less than 9:'
+                  f'{type(new_total_exp)}) : {len(new_total_exp)}')
+
+    @total_dep_exp.deleter
+    def total_dep_exp(self):
+        raise NotImplementedError('Delete is not supported for this attribute')
+
+    @property
+    def salary_of_worker(self):
+        return self.__salary_of_worker
+
+    @salary_of_worker.setter
+    def salary_of_worker(self, new_salary):
+        if isinstance(new_salary, int | float) and len(str(new_salary)) <= 7:
+            self.__salary_of_worker = new_salary
+        else:
+            print(f'Salary of worker should be a int or float with length equal or less than 7:'
+                  f'{type(new_salary)}) : {len(new_salary)}')
+
+    @salary_of_worker.deleter
+    def salary_of_worker(self):
+        raise NotImplementedError('Delete is not supported for this attribute')
+
+    def __str__(self):
+        class_methods = [
+            'workers_quantity_predict_in_2_years',
+            'calculate_middle_sal_of_all_de'
+        ]
+        class_methods_str = ", ".join(class_methods)
+        return f"{self.department_name}, {self.workers_quantity}, {self.total_dep_exp}, {self.productivity_percent_of_department}, " \
+               f"{self.salary_of_worker}, Class Methods: {class_methods_str}"
```

Comment:
I do not mean to override magic methods. I mean just regular functions. 
For example:
def calculate_future_budget(self, ...params...):
       logic for calculating budget